substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+//! - **TemporaryRewards:**For test purposes only ⇒ used as a substitute for the inflation portion
+//!   used for the rewards.
+//! - **PotId:** Pot containing the funds used to pay the rewards.
+//! - **ClaimingPeriod:**Period during which allocated funds can be claimed
+//!
+//! ## Interface
+//!
+//! ### Permissioned Calls
+//! * `register_projects_batch`: Allows a AdminOrigin to register a list of whitelisted projects for
+//!   funding allocation
+//! * `unregister_project`: Allows an AdminOrigin to unregister a previously whitelisted project
+//!
+//! ### Permissionless Calls
+//! * `vote`: Allows users to [vote for/nominate] a whitelisted project using their funds.
+//! * `remove_vote`: Allows users to remove a casted vote.
+//! * `release_voter_funds`: Allows users to unlock funds related to a specific project.
+//! * `claim_reward_for`: Claim a reward for a nominated/whitelisted project.
+//! * `execute_call_dispatch`: Used for delayed calls execution
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+mod functions;
+mod traits;
+mod types;
+pub use traits::*;
+pub use types::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+
+pub use weights::WeightInfo;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeCall: Parameter
+			+ UnfilteredDispatchable<RuntimeOrigin = <Self as frame_system::Config>::RuntimeOrigin>
+			+ From<Call<Self>>
+			+ Into<<Self as frame_system::Config>::RuntimeCall>
+			+ Into<
+				<Self::Governance as traits::ReferendumTrait<
+					<Self as frame_system::Config>::AccountId,
+				>>::Call,
+			> + GetDispatchInfo;
+		/// The admin origin that can list and un-list whitelisted projects.
+		type AdminOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Type to access the Balances Pallet.
+		type NativeBalance: fungible::Inspect<Self::AccountId>
+			+ fungible::Mutate<Self::AccountId>
+			+ fungible::hold::Inspect<Self::AccountId>
+			+ fungible::hold::Mutate<Self::AccountId, Reason = Self::RuntimeHoldReason>
+			+ fungible::freeze::Inspect<Self::AccountId>
+			+ fungible::freeze::Mutate<Self::AccountId>;
+
+		/// The type implementing the `ReferendumTrait` for this pallet.
+		///
+		/// This associated type defines the implementation of referendum-related functionality,
+		/// including creating proposals, submitting them, and managing referendum information.
+		type Governance: ReferendumTrait<Self::AccountId>;
+
+		/// The type implementing the `ConvictionVotingTrait` for this pallet.
+		///
+		/// This associated type defines the implementation of conviction voting functionality,
+		/// including vote creation, submission, removal, and balance unlocking.
+		type Conviction: ConvictionVotingTrait<Self::AccountId>;
```

Comment:
`pallet-conviction-voting` only locks the conviction for the voter that voted same as the result of the poll.
But we want to lock the conviction for both the `Yes` and the `No`.
Also in the spec the referendum are always "passing", to be more precise there is no concept of passing or failing. it is just a poll.

So we need to modify `pallet-conviction-voting` to lock the conviction regardless of the result of the poll. This can be done with an additional associated type in the trait `Config` of `pallet-conviction-voting`.